clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -527,7 +527,15 @@ EmitMaterializeTemporaryExpr(const MaterializeTemporaryExpr *M) {
     // initialized it.
     if (!Var->hasInitializer()) {
       Var->setInitializer(CGM.EmitNullConstant(E->getType()));
-      EmitAnyExprToMem(E, Object, Qualifiers(), /*IsInit*/true);
+      QualType RefType = M->getType();
+      if (RefType.getPointerAuth()) {
```

Comment:
I would suggest just unconditionally dropping the CVR qualifiers from the type of `M`.  There's probably a related ARC bug here, involving either `__weak` or `__strong`?

Reply:
It looks like the code is correct. `__strong` and `__weak` pointers are handled at the beginning of the function.